### PR TITLE
Create database upgrade from 4.11.0.0 to 4.11.1.0 & VMWare version to OS mappings

### PR DESCRIPTION
**Pull request by ernjvr**
Jira ticket: https://issues.apache.org/jira/browse/CLOUDSTACK-10325

Also, add the missing 'VMWare version to OS mappings' SQL in the schema-41100to41110.sql file. This fixes an upgrade failure for certain guest OSs where the Virtual Router fails to start when these version mappings are missing.

Dev Tested the schema-41100to41110.sql file by executing it with the com/cloud/utils/db/ScriptRunner.java class. Before and after data samples confirm a test pass status.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
looks fine @ernjvr let's give it a test souce
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java [hunk 1]
```diff
@@ -120,7 +121,7 @@ public DatabaseUpgradeChecker() {
                 new Upgrade410to420(), new Upgrade420to421(), new Upgrade421to430(), new Upgrade430to440(), new Upgrade440to441(), new Upgrade441to442(), new Upgrade442to450(),
                 new Upgrade450to451(), new Upgrade451to452(), new Upgrade452to460(), new Upgrade460to461(), new Upgrade461to470(), new Upgrade470to471(), new Upgrade471to480(),
                 new Upgrade480to481(), new Upgrade481to490(), new Upgrade490to4910(), new Upgrade4910to4920(), new Upgrade4920to4930(), new Upgrade4930to41000(),
-                new Upgrade41000to41100()});
+                new Upgrade41000to41100(), new Upgrade41100to41110()});
```
Comment (nitin-maharana) [inline engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java]: @ernjvr, Thanks for the PR. Would be good if you add a unit test to find the upgrade path from 41100 to 41110 in this file (engine/schema/test/com/cloud/upgrade/DatabaseUpgradeCheckerTest.java).

**Comment by ernjvr** [comment]
@nitin-maharana thanks. I have added unit test to DatabaseUpgradeCheckerTest.java

**Comment by ernjvr** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65

**Comment by rhtyd** [comment]
@nvazquez can you check if the two failures (one related to cpu cap) were caused by recent changes/prs?

**Comment by ernjvr** [comment]
@nitin-maharana @nvazquez do you approve?

**Comment by ernjvr** [comment]
Upgrade Test Result: Created an environment to install a 4.11 version using this PR code. After the install, I verified in the   /var/log/cloudstack/management/management-server.log file that the Upgrade41100to41110 script did run. Then logged into mysql and ran a SQL query to sample the data to verify that the VMWare version-to-OS mappings are correct. The sample data is correct.

**Comment by DaanHoogland** [comment]
@nitin-maharana @nvazquez @rhtyd can we merge this?

**Comment by nvazquez** [comment]
Sorry for the delay @ernjvr @rhtyd @DaanHoogland. I've examined CPU cap failure and found out that test was marked as successfull even though it ran for 0s, which lead to error as '_cleanup' variable wasn't set: 

````
2018-03-21 15:38:27,715 - DEBUG - ::::::::::::STARTED : TC: test_01_service_offering_cpu_limit_use :::::::::::
2018-03-21 15:38:27,715 - DEBUG - TestCaseName: test_01_service_offering_cpu_limit_use; Time Taken: 0 Seconds; StartTime: Wed Mar 21 15:38:27 2018; EndTime: Wed Mar 21 15:38:27 2018; Result: SUCCESS
````

**Comment by DaanHoogland** [comment]
@nvazquez I don't understand. Does that mean we are good to go with this PR?

**Comment by nvazquez** [comment]
@DaanHoogland yes, errors are not related to this PR. I meant to say that CPU Cap test shouldn't have ran on Vmware as it was defined only for KVM.

**Comment by DaanHoogland** [comment]
so LGTM from you @nvazquez ?

**Comment by nvazquez** [comment]
Yes, LGTM

**Comment by DaanHoogland** [comment]
@nitin-maharana @borisstoyanov are you guys alright with this as is?

**Comment by nitin-maharana** [comment]
Sorry @DaanHoogland, Somehow I missed your message. LGTM from my side. Anyway, it's merged as it had already two LGTMs. Thanks.

**Comment by DaanHoogland** [comment]
np @nitin-maharana , would be if you had objection though. And we can always revert, remember ;)

**Pull request merged**